src/conn/conn_compact.c

```diff
@@ -518,19 +518,23 @@ __background_compact_find_next_uri(WT_SESSION_IMPL *session, WT_ITEM *uri, WT_IT
 static WT_THREAD_RET
 __background_compact_server(void *arg)
 {
+    WT_CACHE *cache;
     WT_CONNECTION_IMPL *conn;
     WT_DECL_ITEM(config);
     WT_DECL_ITEM(next_uri);
     WT_DECL_ITEM(uri);
     WT_DECL_RET;
     WT_SESSION *wt_session;
     WT_SESSION_IMPL *session;
-    bool full_iteration, running;
+    double cache_pressure_limit;
+    bool cache_pressure, full_iteration, running;
 
     session = arg;
     conn = S2C(session);
+    cache = conn->cache;
+    cache_pressure_limit = WT_EVICT_PRESSURE_THRESHOLD * conn->cache_size;
```

Comment:
Why is WT_EVICT_PRESSURE_THRESHOLD a constant?

Reply:
We are checking two things with background compaction:

- The cache is not almost full, hence the usage of `WT_EVICT_PRESSURE_THRESHOLD * conn->cache_size`. I am happy to add `+ 1` here to check for shared cache.
- Dirty eviction is not needed by checking `__wt_eviction_dirty_needed()`

Reply:
If the cache size is not set yet, `cache_pressure_limit` would be 0 which does not make much sense.
I could set `cache_pressure_limit` inside the loop where it's used to make sure it gets set at some point.
But @Siddhartha8899, is that a thing that with shared cache we may have a value of 0 for `conn->cache_size`?

Reply:
After discussing with @Siddhartha8899, he suggested using `__wt_eviction_clean_needed()` as it is an existing function that checks if the cache content is greater than `eviction_trigger`. This is a better solution than using `WT_EVICT_PRESSURE_THRESHOLD` as both `WT_EVICT_PRESSURE_THRESHOLD`  and `eviction_trigger` have a value of 95%. Furthermore, relying on `eviction_trigger` is better for the application as it's a configurable parameter.
Thank you, Sid!
See 09a4b1f7c669f7a314facfc12ba5a3da47a8aca5

Reply:
Should prefetch also use `__wt_eviction_clean_needed` then?

Reply:
Yes, @Siddhartha8899 and I made the same observation.
cc @mm-ng